tests/e2e/util/kubectl/kubectl.go

```diff
@@ -129,18 +131,33 @@ func (k Kubectl) ApplyString(yamlString string) error {
 
 // Apply applies the given yaml file to the cluster
 func (k Kubectl) Apply(yamlFile string) error {
-	err := k.ApplyWithLabels(yamlFile, "")
-	return err
+	return k.applyWithOptions("-f " + yamlFile)
 }
 
 // ApplyWithLabels applies the given yaml file to the cluster with the given labels
 func (k Kubectl) ApplyWithLabels(yamlFile, label string) error {
-	cmd := k.build(" apply " + labelFlag(label) + " -f " + yamlFile)
-	_, err := k.executeCommand(cmd)
-	if err != nil {
-		return fmt.Errorf("error applying yaml: %w", err)
+	return k.applyWithOptions(labelFlag(label) + " -f " + yamlFile)
+}
+
+// ApplyKustomize applies the given kustomization file to the cluster and if labels are provided, adds them as well
+func (k Kubectl) ApplyKustomize(appName string, labels ...string) error {
+	args := []string{"-k", getKustomizeDir(appName)}
+	for _, label := range labels {
+		if label != "" {
+			args = append(args, labelFlag(label))
+		}
 	}
+	return k.applyWithOptions(strings.Join(args, " "))
```

Comment:
nit - this should work, not sure why you changed then reverted it?

```suggestion
	return k.applyWithOptions(args...)
```